core/src/main/java/com/graphhopper/routing/util/CurvatureWeighting.java

```diff
@@ -0,0 +1,138 @@
+package com.graphhopper.routing.util;
+
+import com.graphhopper.storage.GraphHopperStorage;
+import com.graphhopper.storage.NodeAccess;
+import com.graphhopper.util.DistanceCalc;
+import com.graphhopper.util.EdgeIteratorState;
+import com.graphhopper.util.Helper;
+import com.graphhopper.util.PMap;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This Class uses Curvature Data to prefer curvy routes.
+ */
+public class CurvatureWeighting extends PriorityWeighting
+{
+
+    private static final Logger logger = LoggerFactory.getLogger(CurvatureWeighting.class);
+
+    private final FlagEncoder flagEncoder;
+    private final NodeAccess nodeAccess;
+
+    private final DistanceCalc distCalc = Helper.DIST_EARTH;
+
+    public CurvatureWeighting( FlagEncoder flagEncoder, PMap pMap, GraphHopperStorage ghStorage )
+    {
+        super(flagEncoder, pMap);
+        this.flagEncoder = flagEncoder;
+        this.nodeAccess = ghStorage.getNodeAccess();
+    }
+
+    @Override
+    public double getMinWeight( double distance )
+    {
+        return 0.001 * distance;
```

Comment:
This is probably not correct. It needs to be a good approximation to make A* working fast and correct. If it is too small A* will be too slow, if it is bigger then a single value in real world then route results can be incorrect.